openmls/src/messages/mod.rs

```diff
@@ -283,6 +282,19 @@ impl GroupInfo {
     ) -> Result<Self, LibraryError> {
         self.payload.sign(backend, credential_bundle)
     }
+
+    #[cfg(test)]
+    pub(crate) fn into_verifiable_group_info(self) -> VerifiableGroupInfo {
```

Comment:
`into_verifiable_group_info` is used in tests, see [https://github.com/openmls/openmls/pull/1127/files#diff-d1d10ba424715e9e5529dc7069797c74a8d946448d844a9ab0bd994ab2e88352R113] or [https://github.com/openmls/openmls/pull/1127/files#diff-d1d10ba424715e9e5529dc7069797c74a8d946448d844a9ab0bd994ab2e88352R185].

What was done before in the `PublicGroupState` tests was to serialize and deserialize it. But I found this code to be distracting and (if it was for this reason)  (de)serialization should be tested more systematically elsewhere.

Reply:
Sorry, I still don't understand. From my understanding this is a method available throughout the crate during testing only. And it should be only available during tests, because, in real life, clients will deserialize a `VerifiableGroupInfo` and not create it from a `GroupInfo`.

> Is there an integration test covering this? Not sure if I've seen one.

Sorry, again. I don't understand what "this" refers to.

Reply:
As discussed OOB, I've added an integration test to see how a member can join by using `export_group_info`, and `join_by_external_commit`.